src/app/plan/components/Wizard/NameSpaceTable.tsx

```diff
@@ -1,78 +1,38 @@
 /** @jsx jsx */
 import { jsx } from '@emotion/core';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import ReactTable from 'react-table';
 import 'react-table/react-table.css';
-import { connect } from 'react-redux';
-import planOperations from '../../duck/operations';
 import { TextContent, TextList, TextListItem } from '@patternfly/react-core';
 import { css } from '@emotion/core';
 import styled from '@emotion/styled';
-interface IState {
-  page: number;
-  perPage: number;
-  pageOfItems: any[];
-  rows: any;
-  selectAll: any;
-  checked: any;
-}
-interface IProps {
-  sourceCluster: any;
-  sourceClusterNamespaces: any;
-  fetchNamespacesForCluster: any;
-  setFieldValue: (fieldName, fieldValue) => void;
-  values: any;
-}
 
-class NamespaceTable extends React.Component<IProps, IState> {
-  state = {
-    page: 1,
-    perPage: 20,
-    pageOfItems: [],
-    rows: [],
-    checked: [],
-    selectAll: false,
-  };
+const NamespaceTable = props => {
+  const { setFieldValue, sourceClusterNamespaces, values } = props;
+  const [checkedNamespaceRows, setCheckedRows] = useState([]);
 
-  componentDidMount() {
-    const { fetchNamespacesForCluster, sourceCluster, values } = this.props;
-    if (sourceCluster) {
-      fetchNamespacesForCluster(this.props.sourceCluster);
-    }
-  }
 
-  componentDidUpdate(prevProps, prevState) {
-    const { fetchNamespacesForCluster, sourceCluster, values } = this.props;
-    if (prevProps.sourceClusterNamespaces !== this.props.sourceClusterNamespaces) {
-      this.setState({ rows: this.props.sourceClusterNamespaces }, () => {
-        if (values.selectedNamespaces.length > 0) {
-          const { rows } = this.state;
-          const checkedCopy = [];
-          values.selectedNamespaces.filter((item, itemIndex) => {
-            for (let i = 0; rows.length > i; i++) {
-              if (item.metadata.name === rows[i].metadata.name) {
-                checkedCopy[i] = true;
-              }
-            }
-          });
-          this.setState({
-            checked: checkedCopy,
-          });
+  useEffect(() => {
+    if (values.selectedNamespaces.length > 0) {
+      const checkedCopy = [];
+      values.selectedNamespaces.filter((item, itemIndex) => {
+        for (let i = 0; sourceClusterNamespaces.length > i; i++) {
+          if (item.metadata.name === sourceClusterNamespaces[i].metadata.name) {
+            checkedCopy[i] = true;
+          }
         }
       });
+      setCheckedRows(checkedCopy);
     }
-  }
+  }, [sourceClusterNamespaces]);
```

Comment:
The useEffect hook will watch the sourceClusterNamespaces for any changes and run the code in the handler anytime it does change. This replaces the setState callback function that you could link to a setState in class components

Reply:
My concern is that this closure is also referencing values, which is coming off the props. The docs suggest that `values` could change, but this would only fire when `sourceClusterNamespaces` changes, so `values` would be stale in that case.

Reply:
Good catch - I will add values to the array